addons/web/controllers/json2.py

```diff
@@ -0,0 +1,141 @@
+        return request.httprequest.mimetype == 'application/json'
+
+    def dispatch(self, endpoint, args):
+        self.request.params = dict(args)
+        if self.request.httprequest.content_length:
+            try:
+                self.request.params |= self.request.get_json_data()
+            except ValueError as exc:
+                raise BadRequest(f"could not parse the body as json: {exc.args[0]}") from exc
+            except TypeError as exc:
+                # We don't need anything else than json object at the
+                # moment but we are open to discuss supporting the other
+                # data types. Reach us if you have a use case!
+                raise BadRequest("could not parse the body as a json object") from exc
+        if self.request.db:
+            result = self.request.registry['ir.http']._dispatch(endpoint)
+        else:
+            result = endpoint(**self.request.params)
+        if isinstance(result, http.Response):
+            return result
+        return self.request.make_json_response(result)
+
+    def handle_error(self, exc: Exception) -> Callable:
+        headers = None
+        cookies = None
+        if isinstance(exc, UserError):
+            msg = exc.args[0]
+            status = exc.http_code
+        elif isinstance(exc, HTTPException):
+            if res := exc.response:
+                ct = res.headers.get('Content-Type', 'application/octet-stream')
+                if 'json' in ct or not ct.startswith('text/'):
+                    return exc
+                msg = res.get_data(as_text=True)
+                status = exc.code
+                headers = res.headers
+            else:
+                msg = exc.description
+                status = exc.code
+        elif isinstance(exc, http.SessionExpiredException):
+            # needed by auth='user' but not by auth='bearer'
+            msg = (
+                "Session Expired\n\n"
+                "Surely this is because you are using a session_id "
+                "cookie but that session does no longer exists.\n"
+                "Either GET and POST the html form at /web/login?db=\n"
+                "Either POST using JSONRPC your credentials to "
+                "/web/session/authenticate with method 'call' and "
+                "params 'db', 'login', and 'password'.\n"
+                "In either case, your session_id cookie will be "
+                "updated."
+            )
+            status = HTTPStatus.FORBIDDEN
+        else:
+            msg = "internal server error"
+            status = HTTPStatus.INTERNAL_SERVER_ERROR
+        return self.request.make_json_response(msg, headers, cookies, status)
+
+
+class WebJson2Controller(http.Controller):
+    def _web_json_2_rpc_readonly(self, rule, args):
+        try:
+            model_name = args['model']
+            method_name = args['method']
+            Model = request.registry[model_name]
+        except KeyError:
+            return False
```

Comment:
We can use the RO connection for non-existent methods.
```suggestion
            return True
```